adafruit_fona/adafruit_fona.py

```diff
@@ -221,64 +166,114 @@ def _init_fona(self):
                 self._fona_type = FONA_800_H
         return True
 
-    @property
-    def gprs(self):
-        """Returns module's GPRS state."""
+    def factory_reset(self):
+        """Resets modem to factory configuration."""
         if self._debug:
-            print("* Check GPRS State")
-        if not self._send_parse_reply(b"AT+CGATT?", b"+CGATT: ", ":"):
+            print("\t---> ATZ")
+        self._uart.write(b"ATZ\r\n")
+
+        if not self._expect_reply(REPLY_OK):
             return False
-        return self._buf
+        return True
 
-    def configure_gprs(self, config):
-        """If config provided, sets GPRS configuration to provided tuple in format:
-        (apn_network, apn_username, apn_password)
+    def reset(self):
+        """Performs a hardware reset on the modem.
+        NOTE: This may take a few seconds to complete.
 
         """
         if self._debug:
-            print("* Setting GPRS Config to: ", config)
-        apn, username, password = config
-        self._apn = apn.encode()
-        self._apn_username = username.encode()
-        self._apn_password = password.encode()
-        return self._apn, self._apn_username, self._apn_password
-
-    @gprs.setter
-    def gprs(self, gprs_on=True):
-        """Sets GPRS configuration.
-        :param bool gprs_on: Turns on GPRS, enabled by default.
+            print("* Resetting modem.")
+        # Reset the module
+        self._rst.value = True
+        time.sleep(0.01)
+        self._rst.value = False
+        time.sleep(0.1)
+        self._rst.value = True
 
-        """
-        attempts = 5
-        while not self._set_gprs(gprs_on):
-            if attempts == 0:
-                raise RuntimeError("Unable to establish PDP context.")
-            if self._debug:
-                print("* Unable to bringup network, retrying, ", attempts)
-            self._set_gprs(False)
-            attempts -= 1
-            time.sleep(5)
+    @property
+    # pylint: disable=too-many-return-statements
+    def version(self):
+        """Returns FONA Version, as a string."""
+        if self._fona_type == FONA_800_L:
+            return "FONA 800L"
+        if self._fona_type == FONA_800_H:
+            return "FONA 800H"
+        if self._fona_type == FONA_808_V1:
+            return "FONA 808 (v1)"
+        if self._fona_type == FONA_808_V2:
+            return "FONA 808 (v2)"
+        if self._fona_type == FONA_3G_A:
+            return "FONA 3G (US)"
+        if self._fona_type == FONA_3G_E:
+            return "FONA 3G (EU)"
+        return -1
+
+    @property
+    def iemi(self):
+        """Returns FONA module's IEMI number."""
+        self._buf = b""
+        self._uart.reset_input_buffer()
+
+        if self._debug:
```

Comment:
this way of debugging will get tiring. can u have a function that writes to the transport (UART in this case) and have the debut output there

Reply:
@ladyada  sure, that's a better idea.